diff --git a/node_modules/@sentry/replay/cjs/index.js b/node_modules/@sentry/replay/cjs/index.js
index dee41d5..3eb345b 100644
--- a/node_modules/@sentry/replay/cjs/index.js
+++ b/node_modules/@sentry/replay/cjs/index.js
@@ -2020,7 +2020,8 @@ class MutationBuffer {
                         if (attributeName === 'style') {
                             const old = unattachedDoc.createElement('span');
                             if (m.oldValue) {
-                                old.setAttribute('style', m.oldValue);
+                                const s = m.oldValue.split(';').map(i => i.split(':').map(i => i.trim()));
+                                for(const [k, v] of s) { if(k && v) old.style[k] = v };
                             }
                             for (const pname of Array.from(target.style)) {
                                 const newValue = target.style.getPropertyValue(pname);
diff --git a/node_modules/@sentry/replay/esm/index.js b/node_modules/@sentry/replay/esm/index.js
index 83f9ab6..1ab70d1 100644
--- a/node_modules/@sentry/replay/esm/index.js
+++ b/node_modules/@sentry/replay/esm/index.js
@@ -2014,7 +2014,8 @@ class MutationBuffer {
                         if (attributeName === 'style') {
                             const old = unattachedDoc.createElement('span');
                             if (m.oldValue) {
-                                old.setAttribute('style', m.oldValue);
+                                const s = m.oldValue.split(';').map(i => i.split(':').map(i => i.trim()));
+                                for(const [k, v] of s) { if(k && v) old.style[k] = v };
                             }
                             for (const pname of Array.from(target.style)) {
                                 const newValue = target.style.getPropertyValue(pname);
